modules/client/jvm/src/main/scala/com/azavea/stac4s/api/client/SttpStacClient.scala

```diff
@@ -0,0 +1,89 @@
+package com.azavea.stac4s.api.client
+
+import com.azavea.stac4s.{StacCollection, StacItem}
+
+import cats.MonadError
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+import eu.timepit.refined.types.string.NonEmptyString
+import io.chrisdavenport.log4cats.Logger
+import io.circe.Json
+import io.circe.syntax._
+import sttp.client3._
+import sttp.client3.circe.asJson
+import sttp.model.Uri
+
+case class SttpStacClient[F[_]: MonadError[*[_], Throwable]: Logger](
+    client: SttpBackend[F, Any],
+    baseUri: Uri
+) extends StacClient[F] {
+
+  type Filter = SearchFilters
+
+  def search(filter: SearchFilters = SearchFilters()): F[List[StacItem]] =
+    Logger[F].trace(s"search: ${filter.asJson.spaces4}") >>
```

Comment:
@jisantuc How do you feel about having logger here? I have a feeling that probably we can / need to remove it out of here. Logging and tracing can be achieved via `Mid`.

Reply:
I'm pretty ok with it tbh, especially since it's at the trace level 🤷🏻‍♂️

I'm not extremely eager to jump right into `Mid`, since this is the first effectful code of any kind in `stac4s`